### PR TITLE
Added schema hook for setAttr #127

### PR DESCRIPTION
**Pull request by dnalagatla**

### CODE REVIEW

File: tests/unit/projection-test.js [hunk 1]
```diff
@@ -1780,6 +1780,257 @@ module('unit/projection', function(hooks) {
...
+      });
+
+      this.records = {
+        baseRecord,
+        projectedExcerpt,
+        projectedPreview,
+      };
+
+      //Adding .setAttribute hook in schema
+      SchemaManager.schema.setAttribute = function(modelName, attr, value, schemaInterface) {
```
Comment (hjdivad) [inline tests/unit/projection-test.js]: shouldn't we do this before we push anything into the store?

File: tests/unit/projection-test.js [hunk 2]
```diff
@@ -1780,6 +1780,257 @@ module('unit/projection', function(hooks) {
     });
   });
 
+  module('Update projection property with resolved value', function(hooks) {
+    // properties for use for initial state
+    const BOOK_ID = 'isbn:9780439708181';
+    // TODO is this valid? we won't have a real ID yeah?
```
Comment (hjdivad) [inline tests/unit/projection-test.js]: not following this TODO; what's the issue?

**Comment by hjdivad** [comment]
@dnalagatla can we update `README.md` with `setAttribute` as well?

**Comment by hjdivad** [comment]
- [ ] as you observed, we should also remove [this code](https://github.com/hjdivad/ember-m3/blob/40c03c55fc769dcf7bc867d55ae951fa680978cd/addon/model.js#L524-L538) and replace it with a call to the `setAttribute` hook

File: tests/unit/projection-test.js (hunk 2 above)
Comment (dnalagatla) [reply tests/unit/projection-test.js]: Oh, I copied projection update from a another test module, missed removing the TODO. I will fix that.

File: tests/unit/projection-test.js (hunk 1 above)
Comment (dnalagatla) [reply tests/unit/projection-test.js]: That is a good point.

But Looking at the code of `store.push`, it doesn't invoke `.set` method on the model, as result probably I didn't see any error. 

But I will update the code to set the hook before invoking the store.push.

**Pull request merged**